### ISSUE TITLE
Significance of block name?

### ISSUE THREAD

**Issue by dhardy** [gfx-rs/naga#1115]
I comment on this *mostly* because I notice that validation differs between Vulkan and GL backends.

Given two-part shaders, both with uniform blocks:
```
// shader.vert
layout(set = 0, binding = 0) uniform Common {
    vec2 scale;
};
// ...

// shader.frag
layout(set = 0, binding = 1) uniform Common {
    vec3 lightNorm;
};
// ...
```
The Vulkan backend accepts these shaders while the GL backend yields an error:
```
[2021-07-20T14:47:15Z ERROR wgpu::backend::direct] wgpu error: Validation Error
    
    Caused by:
        In Device::create_render_pipeline
          note: label = `SS render_pipeline`
        Internal error in VERTEX | FRAGMENT | VERTEX_FRAGMENT shader: Program 4 linking error:error: definitions of uniform block `Locals' do not match
```
Renaming the `Common` blocks to unique names fixing this error.

Quoting from the GL specification (v4.60, section 4.3.9, emphasis added):

> The block name (block-name) is used to match within shader interfaces: an output block of one
pipeline stage will be matched to an input block with the same name in the subsequent pipeline
stage. **For uniform or shader storage blocks, the application uses the block name to identify the
block. Block names have no other use within a shader beyond interface matching;** it is a compile-
time error to use a block name at global scope for anything other than as a block name (e.g. use of a
block name for a global variable name or function name is currently reserved). It is a compile-time
error to use the same block name for more than one block declaration in the same shader interface
(as defined above) within one shader, even if the block contents are identical.

Since the block name is *not* used for interface matching with WGPU it should not be significant in any way. Or if it must be used internally, then it should be validated the same way on other backends.

**Comment by kvark** [comment]
This is not a validation error. It's an internal error, caused by Naga's GL backend.
It needs to make sure the names don't accidentally intersect. I.e. it could add a suffix to the names, like `CommonVs`, `CommonFs`. cc @Gordon-F

**Comment by dhardy** [comment]
Fixed in #1119 by making the name unique during translation.
